### PR TITLE
Molmo Requirements

### PR DESCRIPTION
**Pull request by Eyshika**
Add Molmo-specific requirements file for stable accuracy on different architectures.
Without these requirements user installs latest libs which doesnt provide same accuracy as it would on required libs.

### CODE REVIEW

**Comment by DarkLight1337** [comment]
Could you elaborate on the accuracy problem? We should try to fix it rather than lock an old version....

**Comment by DarkLight1337** [comment]
In that case, can you add a note to the Supported Models page to direct users to install this requirements file (noting that it uses an older vLLM version)? And probably also open a PR to the model repo on HF Hub to raise the issue.

**Comment by Eyshika** [comment]
> In that case, can you add a note to the Supported Models page to direct users to install this requirements file (noting that it uses an older vLLM version)? And probably also open a PR to the model repo on HF Hub to raise the issue.

Just updated `supported_models.md`. i will also create an issue at HF too

**Comment by DarkLight1337** [comment]
Can you satisfy the DCO check as well by signing off your commits?

File: docs/source/models/supported_models.md [hunk 1]
```diff
@@ -16,6 +16,12 @@ If vLLM natively supports a model, its implementation can be found in <gh-file:v
 
 These models are what we list in <project:#supported-text-models> and <project:#supported-mm-models>.
 
+### AllenAI Molmo-7B-D-0924 (multi-modal)
+
+⚠️ Accuracy Note: For improved output quality (especially in object localization tasks), we recommend using the pinned dependency versions listed in [`requirements/molmo.txt`](https://github.com/vllm-project/vllm/blob/main/requirements/molmo.txt).\
+These versions match the environment that achieved consistent results on different GPUs.\
+_Note: This setup currently works with `vllm==0.7.0`._
+
```
Comment (DarkLight1337) [inline docs/source/models/supported_models.md:24]: Let's put this with the other notes in the multi-modal models section

**Comment by Eyshika** [comment]
> Can you satisfy the DCO check as well by signing off your commits?

Ready now!

File: docs/source/models/supported_models.md [hunk 2]
```diff
@@ -124,10 +130,10 @@ class MyConfig(PretrainedConfig):
 
 - `base_model_tp_plan` is a `dict` that maps fully qualified layer name patterns to tensor parallel styles (currently only `"colwise"` and `"rowwise"` are supported).
 - `base_model_pp_plan` is a `dict` that maps direct child layer names to `tuple`s of `list`s of `str`s:
-  * You only need to do this for layers which are not present on all pipeline stages
-  * vLLM assumes that there will be only one `nn.ModuleList`, which is distributed across the pipeline stages
-  * The `list` in the first element of the `tuple` contains the names of the input arguments
-  * The `list` in the last element of the `tuple` contains the names of the variables the layer outputs to in your modeling code
+  - You only need to do this for layers which are not present on all pipeline stages
```
[the remark this answers is not in the document]
Comment (Eyshika) [reply docs/source/models/supported_models.md]: Updated

File: docs/source/models/supported_models.md [hunk 3]
```diff
@@ -776,6 +776,12 @@ See [this page](#multimodal-inputs) on how to pass multi-modal inputs to the mod
 **To enable multiple multi-modal items per text prompt in vLLM V0**, you have to set `limit_mm_per_prompt` (offline inference)
 or `--limit-mm-per-prompt` (online serving). For example, to enable passing up to 4 images per text prompt:
 
+### AllenAI Molmo-7B-D-0924 (multi-modal)
+
+⚠️ Accuracy Note: For improved output quality (especially in object localization tasks), we recommend using the pinned dependency versions listed in [`requirements/molmo.txt`](https://github.com/vllm-project/vllm/blob/main/requirements/molmo.txt).  
+These versions match the environment that achieved consistent results on both A10 and L40 GPUs.  
+_Note: This setup currently works with `vllm==0.7.0`._
+
```
Comment (DarkLight1337) [inline docs/source/models/supported_models.md:784]: Can you move this after the table? Where the other notes are.

Comment (Eyshika) [reply docs/source/models/supported_models.md:784]: Do you mean inside `Model Support Policy` ?

Comment (Eyshika) [reply docs/source/models/supported_models.md]: If yes, then done!

Comment (DarkLight1337) [reply docs/source/models/supported_models.md]: No, I mean near L1084

Comment (Eyshika) [reply docs/source/models/supported_models.md]: ok done

File: docs/source/models/supported_models.md [hunk 4]
```diff
@@ -1081,6 +1081,12 @@ Pan-and-scan image pre-processing is currently supported on V0 (but not V1).
 You can enable it by passing `--mm-processor-kwargs '{"do_pan_and_scan": True}'`.
 :::
 
+### AllenAI Molmo-7B-D-0924 (multi-modal)
+
+⚠️ Accuracy Note: For improved output quality (especially in object localization tasks), we recommend using the pinned dependency versions listed in [`requirements/molmo.txt`](https://github.com/vllm-project/vllm/blob/main/requirements/molmo.txt).  
+These versions match the environment that achieved consistent results on both A10 and L40 GPUs.  
+_Note: This setup currently works with `vllm==0.7.0`._
+
```
[the remark this answers is not in the document]
Comment (Eyshika) [reply docs/source/models/supported_models.md:1090]: done

[the remark this answers is not in the document]
Comment (DarkLight1337) [reply docs/source/models/supported_models.md:1090]: How about alphabetical ordering?

File: docs/source/models/supported_models.md [hunk 5]
```diff
@@ -1113,6 +1107,14 @@ This limitation exists because the model's mixed attention pattern (bidirectiona
 To use `TIGER-Lab/Mantis-8B-siglip-llama3`, you have to pass `--hf_overrides '{"architectures": ["MantisForConditionalGeneration"]}'` when running vLLM.
 :::
 
+:::{warning}
+AllenAI Molmo-7B-D-0924 (multi-modal)
+
+⚠️ Accuracy Note: For improved output quality (especially in object localization tasks), we recommend using the pinned dependency versions listed in [`requirements/molmo.txt`](https://github.com/vllm-project/vllm/blob/main/requirements/molmo.txt).  
+These versions match the environment that achieved consistent results on both A10 and L40 GPUs.  
+_Note: This setup currently works with `vllm==0.7.0`._
```
Comment (DarkLight1337) [inline docs/source/models/supported_models.md:1115]:
```suggestion
For improved output quality of `AllenAI/Molmo-7B-D-0924` (especially in object localization tasks), we recommend using the pinned dependency versions listed in <gh-file:requirements/molmo.txt> (including `vllm==0.7.0`). These versions match the environment that achieved consistent results on both A10 and L40 GPUs.
```

Finally some nits

**Review by DarkLight1337 — APPROVED** [review]
Thanks for bearing with me. LGTM now!

**Comment by russellb** [comment]
https://github.com/vllm-project/vllm/pull/17189

**Comment by DarkLight1337** [comment]
I have opened another PR that doesn't revert this one completely but should avoid the security flags you've mentioned: https://github.com/vllm-project/vllm/pull/17190
